src/utils/mod.rs

```diff
@@ -435,7 +435,7 @@ pub(crate) fn delete_dir_contents_following_links(dir_path: &Path) {
 
     match raw::open_dir_following_links(dir_path).and_then(|mut p| p.remove_dir_contents(None)) {
         Err(e) if e.kind() != io::ErrorKind::NotFound => {
-            panic!("Unable to clean up {}: {:?}", dir_path.display(), e);
+            warn!("unable to clean up {}: {e:?}", dir_path.display());
```

Comment:
```suggestion
            warn!("unable to clean up {}: {e}", dir_path.display());
```

I think it'd be nicer to use the `Display` formatting for the error.

Reply:
I'm not 100% sure about that since it might still be interesting to look at the underlying error in this case, and the `Display` implementation only focuses on the error message...

```rs
fn main() -> std::io::Result<()> { std::fs::File::open("foo.txt").map(drop) }
```

... this gives:

```console
> cargo run
[..]
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
```

Returning to the example of #4609, `Os { code: 5, kind: PermissionDenied, message: "Access is denied." }` has also made the origin of the error much clearer IMHO. Otherwise it might not be immediately clear where the `Access is denied.` string is coming from?

Reply:
Hm, the difference is:

Before:
```
Os { code: 5, kind: PermissionDenied, message: "Access is denied." }
```

After:
```
Access is denied. (os error 5)
```

I think the latter looks nicer and the only thing it's missing is the the error kind.

Reply:
@ChrisDenton Okay, I'm convinced. Many thanks 🙏